relayer/path.go

```diff
@@ -85,14 +87,30 @@ type PathAction struct {
 	Type string `json:"type"`
 }
 
-// Path represents a pair of chains and the identifiers needed to
-// relay over them
+// Path represents a pair of chains and the identifiers needed to relay over them along with a channel filter list.
 type Path struct {
-	Src *PathEnd `yaml:"src" json:"src"`
-	Dst *PathEnd `yaml:"dst" json:"dst"`
+	Src    *PathEnd       `yaml:"src" json:"src"`
+	Dst    *PathEnd       `yaml:"dst" json:"dst"`
+	Filter *ChannelFilter `yaml:"src-channel-filter" json:"src-channel-filter"`
 }
 
-// End returns the proper end given a chainID
+// ChannelFilter provides the means for either creating an allowlist or a denylist of channels on the src chain
+// which will be used to narrow down the list of channels a user wants to relay on.
+type ChannelFilter struct {
+	Rule        string   `yaml:"rule" json:"rule"`
+	ChannelList []string `yaml:"channel-list" json:"channel-list"`
+}
+
+// ValidateChannelFilterRule verifies that the configured ChannelFilter rule is set to an appropriate value.
+func (p *Path) ValidateChannelFilterRule() error {
+	if p.Filter.Rule != allowList && p.Filter.Rule != denyList && p.Filter.Rule != "" {
+		return fmt.Errorf("%s is not a valid channel filter rule, please "+
+			"ensure your channel filter rule is `allowlist` or 'denylist'", p.Filter.Rule)
```

Comment:
Lets be really clear on this error. 
"ensure your channel filter rule is either `allowlist` or `denylist`, not both

Reply:
Or something like that.

Reply:
The way I tweaked this now ensures that the user is only using one or the other since we just provide an option for the filter `rule` type (e.g. `allowlist` or `denylist`) and a single list of channels.

Reply:
optional right?